app/components/insights-glance.js

```diff
@@ -0,0 +1,87 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { or } from '@ember/object/computed';
+import { format as d3format } from 'd3';
+
+export default Component.extend({
+  classNames: ['insights-glance'],
+  classNameBindings: ['isLoading:insights-glance--loading'],
+
+  isLoading: true,
+  isEmpty: false,
+
+  title: '',
+  statistic: '',
+
+  delta: 0,
+  deltaTitle: '',
+  deltaText: '',
+
+  labels: [],
+  values: [],
```

Comment:
Properties of anything inherited from `Ember.Object` or `Ember.Mixin` (so basically literally anything in Ember 😆) should never be initialized by values of complex types (arrays, objects), because it may cause leaking state between instances. Here are more details on the topic: https://dockyard.com/blog/2015/09/18/ember-best-practices-avoid-leaking-state-into-factories

The idea is to either set defaults to such properties in `init` or use computed properties, e.g.:
```javascript
values: computed(() => [])
```

Reply:
Ahh okay, that's good to know! Thanks for the explanation and link, that makes a lot of sense. I'll go ahead and wrap these in computed properties 😄